rust/src/util/bit_util.rs

```diff
@@ -44,13 +44,22 @@ fn round_upto_power_of_2(num: i64, factor: i64) -> i64 {
 /// Returns whether bit at position `i` in `data` is set or not
 #[inline]
 pub fn get_bit(data: &[u8], i: i64) -> bool {
-    (data[(i / 8) as usize] & BIT_MASK[(i % 8) as usize]) != 0
+    (data[(i >> 3) as usize] & BIT_MASK[(i & 7) as usize]) != 0
+}
+
+/// Returns whether bit at position `i` in `data` is set or not.
+///
+/// Note this doesn't do any bound checking, for performance reason. The caller is
+/// responsible to guarantee that `i` is within bounds.
+#[inline]
+pub fn get_bit_raw(data: *const u8, i: i64) -> bool {
+    unsafe { (*data.offset((i >> 3) as isize) & BIT_MASK[(i & 7) as usize]) != 0 }
 }
 
 /// Sets bit at position `i` for `data`
 #[inline]
 pub fn set_bit(data: &mut [u8], i: i64) {
-    data[(i / 8) as usize] |= BIT_MASK[(i % 8) as usize]
+    data[(i >> 3) as usize] |= BIT_MASK[(i & 7) as usize]
```

Comment:
Same as above: what happens if i is negative?

Reply:
Same above. Will change this to `usize` in a separate PR (since it's a broader change).